packages/@aws-cdk/core/lib/bundling.ts

```diff
@@ -193,6 +199,21 @@ export class BundlingDockerImage {
   }
 }
 
+/**
+ * A Docker image
+ */
+export class DockerImage extends BundlingDockerImage {
```

Comment:
When deprecating, I usually prefer that the implementation will be in the non-deprecated class while the deprecated class just extends it (reverse).

Reply:
Yeah me too and I tried that but it's a breaking change because the static methods would return an instance of `DockerImage` which doesn't extend `BundlingDockerImage`.